### PR TITLE
fix EditUserRequestValidator

### PR DESCRIPTION
**Pull request by Cheephot**

### CODE REVIEW

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 1]
```diff
@@ -1,376 +1,304 @@
...
-                    { x => bool.TryParse(x.value?.ToString(), out _), "Incorrect user is active format" }
-                });
+        nameof(EditUserRequest.DateOfBirth),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+        {
+          {
+            x => string.IsNullOrEmpty(x.value?.ToString())? true :
+              DateTime.TryParse(x.value.ToString(), out DateTime result),
+            "Date of birth has incorrect format"
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:196]: "Date of birth has incorrect format."

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 2]
```diff
@@ -1,376 +1,304 @@
...
+
+      AddFailureForPropertyIf(
+        nameof(EditUserRequest.StartWorkingAt),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+        {
+          {
+            x => string.IsNullOrEmpty(x.value?.ToString())? true :
+              DateTime.TryParse(x.value.ToString(), out DateTime result),
+            "Start working at has incorrect format"
```
Comment (MariyaSin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:212]: "Start working at has incorrect format."

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 1 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 2 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 3]
```diff
@@ -1,376 +1,283 @@
...
-      #region IsActive
+      #region firstname
+
+      AddFailureForPropertyIf(
+        nameof(EditUserRequest.FirstName),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+        {
+          { x => !string.IsNullOrEmpty(x.value?.ToString().Trim()), "First name must not be empty." },
+          { x => !NameRegex.IsMatch(x.value.ToString()), "First name must not contain numbers." },
```
Comment (VLiamin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:76]: If x.value = null, here will be exception.  may be

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 4]
```diff
@@ -1,376 +1,283 @@
...
-                {
-                    { x => bool.TryParse(x.value?.ToString(), out _), "Incorrect user is active format" }
-                });
+        nameof(EditUserRequest.Rate),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+        {
+          { x => !string.IsNullOrEmpty(x.value?.ToString()), "Rate must not be empty." },
+          { x => double.TryParse(x.value?.ToString(), out _), "Incorrect rate format."},
+          { x => double.TryParse(x.value?.ToString(), out double rate) ? (rate > 0 && rate <=1) : false, "The rate must be between 0 and 1." },
```
Comment (VLiamin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:160]: space

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 5]
```diff
@@ -1,376 +1,283 @@
...
+
+      AddFailureForPropertyIf(
+        nameof(EditUserRequest.AvatarFileId),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditUserRequest>, bool>, string>
+        {
+          { x =>
+            string.IsNullOrEmpty(x.value?.ToString())
+              || (Guid.TryParse(x.value?.ToString(), out Guid avatarId)
+                && _imageRepository.Get(new List<Guid> { avatarId }).Any()),
```
Comment (VLiamin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:207]: Is a tab needed here?

**Review by VLiamin — APPROVED** [review]

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 3 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs:76]: We use CascadeMode.Stop, if the value is null, the validator will stop at line 75.

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 5 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs:207]: yes, becose it: a || (b && c)

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 4 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs]: done

File: src/UserService.Validation/User/EditUserRequestValidator.cs [hunk 6]
```diff
@@ -157,7 +157,10 @@ private void HandleInternalPropertyValidation(Operation<EditUserRequest> request
         {
           { x => !string.IsNullOrEmpty(x.value?.ToString()), "Rate must not be empty." },
           { x => double.TryParse(x.value?.ToString(), out _), "Incorrect rate format."},
-          { x => double.TryParse(x.value?.ToString(), out double rate) ? (rate > 0 && rate <=1) : false, "The rate must be between 0 and 1." },
+          { x => double.TryParse(x.value?.ToString(), out double rate) ?
+              (rate > 0 && rate <=1) : false,
```
Comment (VLiamin) [inline src/UserService.Validation/User/EditUserRequestValidator.cs:161]: space

**Review by MariyaSin — APPROVED** [review]

File: src/UserService.Validation/User/EditUserRequestValidator.cs (hunk 6 above)
Comment (Cheephot) [reply src/UserService.Validation/User/EditUserRequestValidator.cs:161]: done

**Review by Glunky — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**